tests/semantic_analyser.cpp

```diff
@@ -538,6 +538,10 @@ TEST(semantic_analyser, call_stats)
 TEST(semantic_analyser, call_delete)
 {
   test("kprobe:f { @x = 1; delete(@x); }", 0);
+  test("kprobe:f { @x = 1; @y = 2; delete(@x, @y); }", 0);
+  test("kprobe:f { @x = 1; @y[5] = 5; delete(@x, @y[5]); }", 0);
+  test("kprobe:f { @x = 1; @y[5] = 5; delete(@x, @y); }", 1);
+  test("kprobe:f { @x = 1; $y = 2; delete(@x, $y); }", 1);
```

Comment:
Another instance of less-than-ideal old tests being bad examples - sorry!

I'm trying to make sure all new semantic analyser tests follow the new pattern of checking the error messages instead of the return code, e.g.: https://github.com/bpftrace/bpftrace/blob/b1dd3706e81d635a342dd37a0530467ee2eb8a9f/tests/semantic_analyser.cpp#L286-L290

Checking the error would be useful for this test specifically, because it currently gives this message:
```
stdin:1:28-42: ERROR: delete() only expects maps to be provided
kprobe:f { @x = 1; $y = 2; delete(@x, $y); }
                           ~~~~~~~~~~~~~~
```

But I think that when there are lots of arguments to `delete()` it'd be more useful to highlight the particular one which is causing the problem:
```
stdin:1:39-41: ERROR: delete() only expects maps to be provided
kprobe:f { @x = 1; $y = 2; delete(@x, $y); }                                                   
                                      ~~
```